projects/timer_fw102/src/main.c

```diff
@@ -0,0 +1,46 @@
+#include <stdbool.h>
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timing_callback(SoftTimerId timer_id, void *context) {
+  Counters *counters = context;
+
+  counters->counter_a++;  // increments a
+
+  LOG_DEBUG("Counter A: %i\n", counters->counter_a);
+
+  if (counters->counter_a % 2 == 0) {  // increments b
+    counters->counter_b++;
+    LOG_DEBUG("Counter B: %i\n", counters->counter_b);
+  }
+
+  // starts timer again
+  soft_timer_start_millis(500, prv_timing_callback, counters, NULL);
```

Comment:
It's good practice to factor out the 500ms constant here and below into a #define'd constant so we're not repeating the constant and it's easier to change if necessary. Make sure to give it a good descriptive name and suffix it with the unit (like `_MS`) so we know what the unit is.